packages/gatsby/src/schema/infer/get-infer-config.js

```diff
@@ -0,0 +1,49 @@
+// @flow
+
+const { Kind } = require(`graphql`)
+
+export interface InferConfig {
+  infer: boolean;
+  addDefaultResolvers: boolean;
+}
+
+const DEFAULT_INFER_CONFIG: InferConfig = {
+  infer: true,
+  addDefaultResolvers: true,
+}
+
+// Get inferance config from type directives
+const getInferConfig: (
+  typeComposer: TypeComposer
+) => InferConfig = typeComposer => {
+  const type = typeComposer.getType()
+  const inferConfig = DEFAULT_INFER_CONFIG
+  if (type.astNode && type.astNode.directives) {
+    type.astNode.directives.forEach(directive => {
+      if (directive.name.value === `infer`) {
+        inferConfig.infer = true
+        inferConfig.addDefaultResolvers = getNoDefaultResolvers(directive)
+      } else if (directive.name.value === `dontInfer`) {
+        inferConfig.infer = false
+        inferConfig.addDefaultResolvers = getNoDefaultResolvers(directive)
+      }
+    })
+  }
+
+  return inferConfig
+}
+
+module.exports = getInferConfig
+
+const getNoDefaultResolvers = directive => {
+  const noDefaultResolvers = directive.arguments.find(
+    ({ name }) => name.value === `noDefaultResolvers`
+  )
+  if (noDefaultResolvers) {
+    if (noDefaultResolvers.value.kind === Kind.BOOLEAN_NODE) {
```

Comment:
I think it's [`Kind.BOOLEAN`](https://github.com/graphql/graphql-js/blob/4116e2fc4fe36688f683258388f4a2d52076d199/src/language/kinds.js#L35)?